docs/en_US/Compressor/Pruner.md

```diff
@@ -390,5 +395,170 @@ You can view example for more information
+- **trainer:** Function used for the first optimization subproblem.
+This function should include `model, optimizer, criterion, epoch, callback` as parameters, where callback should be inserted after loss.backward of the normal training process.
+- **optimize_iteration:** ADMM optimize iterations.
+- **training_epochs:** training epochs of the first optimization subproblem.
+- **row:** penalty parameters for ADMM training.
+- **base_algo:** base pruning algorithm. 'level', 'l1' or 'l2', by default 'l1'.
+            
+
+## AutoCompress Pruner
+For each round t, AutoCompressPruner prune the model for the same sparsity each round to achive the ovrall sparsity:
+        1. Generate sparsities distribution using SimualtedAnnealingPruner
+        2. Perform ADMM-based structured pruning to generate pruning result, for the next round t
+
+Perform prurification step (the speedup process in nni)
+
+For more details, please refer to [AutoCompress: An Automatic DNN Structured Pruning Framework for Ultra-High Compression Rates](https://arxiv.org/abs/1907.03141).
+
+#### Usage
+
+PyTorch code
+
+```python
+from nni.compression.torch import ADMMPruner
+config_list = [{
+        'sparsity': 0.5,
+        'op_types': ['Conv2d']
+    }]
+pruner = AutoCompressPruner(
+            model, config_list, trainer=trainer, evaluator=evaluator,
+            dummy_input=dummy_input, iterations=3, optimize_mode='maximize', base_algo='l1',
+            cool_down_rate=0.9, optimize_iterations=30, epochs=5, experiment_data_dir='./')
+pruner.compress()
+```
+
+You can view [example](https://github.com/microsoft/nni/blob/master/examples/model_compress/auto_pruners_torch.py) for more information.
 
- 
\ No newline at end of file
+#### User configuration for AutoCompress Pruner
+
+- **sparsity:** How much percentage of convolutional filters are to be pruned.
+- **op_types:** "Conv2d" or "default".
+- **trainer:** Function used for the first optimization subproblem.
+This function should include `model, optimizer, criterion, epoch, callback` as parameters, where callback should be inserted after loss.backward of the normal training process.
+- **evaluator:** Function to evaluate the masked model. This function should include `model` as the only parameter, and returns a scalar value.
+- **dummy_input:** The dummy input for model speed up, users should put it on right device before pass in.
+- **iterations:** The number of overall iterations.
+- **optimize_mode:** Optimize mode, 'maximize' or 'minimize', by default 'maximize'.
+- **base_algo:** base pruning algorithm. 'level', 'l1' or 'l2', by default 'l1'.
+- **start_temperature:** Simualated Annealing related parameter.
+- **stop_temperature:** Simualated Annealing related parameter.
+- **cool_down_rate:** Simualated Annealing related parameter.
+- **perturbation_magnitude:** Initial perturbation magnitude to the sparsities. The magnitude decreases with current temperature.
+- **optimize_iteration:** ADMM optimize iterations.
```

Comment:
what is the relation with ADMM?

Reply:
AutoCompress Pruner call `SimualtedAnnealing Pruner`  and `ADMM Pruner ` iteratively.